AK/Time.h

```diff
@@ -56,24 +56,40 @@ constexpr int day_of_year(int year, unsigned month, int day)
 }
 
 // Month starts at 1. Month must be >= 1 and <= 12.
-int days_in_month(int year, unsigned month);
+int days_in_month(i64 year, unsigned month);
 
-constexpr int days_in_year(int year)
+constexpr int days_in_year(i64 year)
 {
     return 365 + (is_leap_year(year) ? 1 : 0);
 }
 
-constexpr int years_to_days_since_epoch(int year)
+constexpr i64 years_to_days_since_epoch(i64 year)
 {
-    int days = 0;
-    for (int current_year = 1970; current_year < year; ++current_year)
-        days += days_in_year(current_year);
-    for (int current_year = year; current_year < 1970; ++current_year)
-        days -= days_in_year(current_year);
-    return days;
```

Comment:
The LHS code here is much shorter and easier to understand. Can't we keep that?

Reply:
The issue here is that the current algorithm is linear-time, and ends up producing **huge** run times when called with large values.

As per the (unsuccessfully) linked issue, some calculations performed by `configure` scripts take more than a minute.

Reply:
In fact, you might not even be able to run the LHS in a constexpr context given a large enough value without changing the compiler settings.

Reply:
What about something like that:
```cpp 
i64 leap_years = year / 400 + year / 4 - year / 1000;
i64 non_leap_years = year - leap_years;
i64 days_since_bce = non_leap_years * 365 + leap_years * 366;
i64 days_from_bce_to_epoch = 719527;
return days_since_bce - days_from_bce_to_epoch;
```